### PR TITLE
pkg: convert substr() usage to substring()

### PR DESCRIPTION
**Pull request by jelly**
String.prototype.substr() is a deprecated API.

---

:see_no_evil: hopefully the retry flakes are not too bad.

### CODE REVIEW

**Review by martinpitt — APPROVED** [review]
Oops, good one! https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substr

I did a round of retries, but supposedly you can/have to land this a little red-ish.

File: pkg/lib/cockpit-path.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ function drop_slashes(path : string): string {
     let pos = path.length;
     while (pos > 1 && path[pos - 1] == "/")
         pos -= 1;
-    return pos == path.length ? path : path.substr(0, pos);
+    return pos == path.length ? path : path.substring(0, pos);
```
Comment (cockpituous) [inline pkg/lib/cockpit-path.ts:25]: This added line is not executed by any test.

File: pkg/lib/cockpit/_internal/base64.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ export function base64_encode(data) {
         }
     }
 
-    return str.substr(0, str.length - 2 + mod3) + (mod3 === 2 ? '' : mod3 === 1 ? '=' : '==');
+    return str.substring(0, str.length - 2 + mod3) + (mod3 === 2 ? '' : mod3 === 1 ? '=' : '==');
```
Comment (cockpituous) [inline pkg/lib/cockpit/_internal/base64.js:33]: This added line is not executed by any test.

File: pkg/packagekit/autoupdates.jsx [hunk 3]
```diff
@@ -253,7 +253,7 @@ class Dnf5Impl extends ImplBase {
             // if we have OnCalendar=, use that (we disable OnUnitInactiveSec= in our drop-in)
             const calIdx = output.indexOf("OnCalendar=");
             if (calIdx >= 0) {
-                this.parseCalendar(output.substr(calIdx).split('\n')[0].split("=")[1]);
+                this.parseCalendar(output.substring(calIdx).split('\n')[0].split("=")[1]);
```
Comment (cockpituous) [inline pkg/packagekit/autoupdates.jsx:256]: This added line is not executed by any test.

File: pkg/systemd/overview-cards/configurationCard.jsx [hunk 4]
```diff
@@ -239,9 +239,9 @@ const PageSystemInformationChangeHostname = () => {
                     .toLowerCase()
                     .replace(/['".]+/g, "")
                     .replace(/[^a-zA-Z0-9]+/g, "-");
-            new_hostname = new_hostname.substr(0, 64);
+            new_hostname = new_hostname.substring(0, 64);
             if (first_dot >= 0)
-                new_hostname = new_hostname + old_hostname.substr(first_dot);
+                new_hostname = new_hostname + old_hostname.substring(first_dot);
```
Comment (cockpituous) [inline pkg/systemd/overview-cards/configurationCard.jsx:244]: This added line is not executed by any test.

**Comment by jelly** [comment]
arch/ubuntu failures are known flakes, merging.

**Pull request merged**